src/main/fc/rc.c

```diff
@@ -704,10 +604,80 @@ static FAST_CODE_NOINLINE void calculateFeedforward(const pidRuntime_t *pid, fli
 }
 #endif // USE_FEEDFORWARD
 
+bool shouldUpdateSmoothing(void)
+{
+    static int validCount = 0;
+    static int outlierCount = 0;
+    static const float smoothingFactor = 0.1f;  // Low pass smoothing factor to smooth valid RxRate values
+    static float lastStableSmoothedRxRateHz = 0.0f;  // stable value to compare currentRxRate against
+
+    const bool ready = (targetPidLooptime > 0);
+
+    if (ready && isRxReceivingSignal() && isRxRateValid) {
```

Comment:
Strongly recommend removing the `ready` check altogether.  It's just a waste of CPU cycles.
I've tested removing it, and there is no difference in the logs.
- the correct link frequency is still stored in the log header.
- the initial frequency on arming is perfectly good.
- the initial frequency on powering up and arming as soon as possible is also perfectly good.
- even if the initial frequency was incorrect, it cannot be extreme (must still be 'valid')
- even if the initial frequency is very wrong, the code quickly homes in on the correct frequency.

Reply:
Great,  nice to see it removed!